src/lib/builders/tree/create.ts

```diff
@@ -250,86 +213,134 @@ export function createTreeView(args?: CreateTreeViewProps) {
 	});
 
 	const group = builder(name('group'), {
-		stores: [expanded],
-		returned: ([$expanded]) => {
-			return (opts: { id: string }) => ({
-				role: 'group',
-				'data-group-id': opts.id,
-				hidden: !forceVisible && !$expanded.includes(opts.id) ? true : undefined,
-				style: styleToString({
-					display: !forceVisible && !$expanded.includes(opts.id) ? 'none' : undefined,
-				}),
-			});
+		stores: [expanded, forceVisible],
+		returned: ([$expanded, $forceVisible]) => {
+			return (opts: { id: string }) => {
+				const { id } = opts;
+				const hidden = !$forceVisible && !$expanded.includes(id);
+				return {
+					role: 'group',
+					id,
+					hidden: hidden ? true : undefined,
+					style: styleToString({
+						display: hidden ? 'none' : undefined,
+					}),
+				} as const;
+			};
 		},
 	});
 
-	function setFocusedItem(el: HTMLElement) {
-		lastFocusedId.update((p) => el.getAttribute('data-id') ?? p);
-		el.focus();
-	}
+	function handleSelection(el: HTMLElement) {
+		if (!el.id) return;
 
-	function updateSelectedElement(el: HTMLElement) {
-		const id = el.getAttribute(ATTRS.DATAID);
-		if (!id) return;
+		if (multiple.get()) {
+			toggleSelection(el);
+		} else if (firstSelected.get() !== el.id) {
+			selected.set([el.id]);
+		}
+	}
 
-		selectedItem.set(el);
+	function toggleSelection(el: HTMLElement) {
+		if (elementIsSelected(el)) {
+			selected.update((prev) => prev.filter((id) => id !== el.id));
+		} else {
+			selected.update((prev) => [...prev, el.id]);
+		}
 	}
 
 	function getItems(): HTMLElement[] {
-		let items = [] as HTMLElement[];
-		const rootEl = getElementByMeltId(metaIds.tree);
-		if (!rootEl) return items;
+		const rootEl = document.getElementById(ids.tree.get());
+		if (!rootEl) return [];
 
 		// Select all 'treeitem' li elements within our root element.
-		items = Array.from(rootEl.querySelectorAll('[role="treeitem"]')).filter(
-			(el) => !isHidden(el as HTMLElement)
-		) as HTMLElement[];
+		return Array.from(rootEl.querySelectorAll('[role="treeitem"]')).filter(
+			(el): el is HTMLElement => isHTMLElement(el) && !isHidden(el)
+		);
+	}
+
+	function focusItemStartingWith(letter: string, items: HTMLElement[], nodeIdx: number) {
```

Comment:
This can be moved outside the closure so the function is only created once rather than on each instantiation since it receives everything it needs as args.